### PR TITLE
Configure pnpm v8+ for Netlify

### PR DESCRIPTION
**Pull request by ProchaLu**
This PR adds a `packageManager` option to the `package.json` file for projects that use `@upleveled/react-scripts` and do not already have a `packageManager` specified. This option is set to the latest version of pnpm, fetched from the npm registry API. Netlify currently uses `pnpm v7`, which can cause issues with deer dependencies. Adding this option will ensure that the correct version of pnpm is used for deployment.

This is the error message when deploying on Netlify without specifying the `packageManager`: 
```
10:07:47 AM: Failed during stage 'Install dependencies': dependency_installation script returned non-zero exit 
10:07:47 AM: hint: If you want peer dependencies to be automatically installed, add "auto-install-peers=true" to an .npmrc file at the root of your project.
10:07:47 AM: hint: If you don't want pnpm to fail on peer dependency issues, add "strict-peer-dependencies=false" to an .npmrc file at the root of your project.
10:07:47 AM: Error during pnpm install
10:07:47 AM: Build was terminated: dependency_installation script returned non-zero exit code: 1
10:07:47 AM: Failing build: Failed to install dependencies
10:07:47 AM: Finished processing build request in 33.255s
```
For more details on the possibility of upgrading to the version of pnpm used in the Netlify build image, visit the following link: https://answers.netlify.com/t/upgrade-netlify-build-to-pnpm-v8/90570.

### CODE REVIEW

File: bin/install.js [hunk 1]
```diff
@@ -222,6 +222,24 @@ writeFileSync(
 
 console.log('✅ Done updating .gitignore');
 
+if (
+  '@upleveled/react-scripts' in projectDependencies &&
+  !projectPackageJson.packageManager
+) {
+  try {
+    const response = await fetch('https://registry.npmjs.org/pnpm/latest');
```
Comment (karlhorky) [inline bin/install.js:230]: let's avoid the fetch and just hardcode it

File: bin/install.js [hunk 2]
```diff
@@ -222,6 +222,24 @@ writeFileSync(
...
+    const data = await response.json();
+    projectPackageJson.packageManager = `pnpm@${data.version}`;
+    writeFileSync(
```
Comment (karlhorky) [inline bin/install.js:233]: we already do a `writeFileSync` to `projectPackageJsonPath`, can we avoid writing to the file twice?

File: bin/install.js (hunk 1 above)
Comment (ProchaLu) [reply bin/install.js]: hardcoded the version number
[447871e](https://github.com/upleveled/eslint-config-upleveled/pull/228/commits/447871e576b1b6843f587bc657575fb8f7a668c6)

File: bin/install.js (hunk 2 above)
Comment (ProchaLu) [reply bin/install.js]: changed the file and only write the file once
[447871e](https://github.com/upleveled/eslint-config-upleveled/pull/228/commits/447871e576b1b6843f587bc657575fb8f7a668c6)

File: bin/install.js [hunk 3]
```diff
@@ -30,6 +30,13 @@ if ('next' in projectDependencies) {
     );
   }
 }
+if (
```
Comment (karlhorky) [inline bin/install.js:33]: let's add:

- an extra line here
- a comment above the `if` to describe why we need this (mention Netlify)

File: bin/install.js [hunk 4]
```diff
@@ -30,6 +30,13 @@ if ('next' in projectDependencies) {
...
+  '@upleveled/react-scripts' in projectDependencies &&
+  !projectPackageJson.packageManager
+) {
+  projectPackageJson.packageManager = 'pnpm@8.3.1';
+  console.log('✅ Added "packageManager" option to package.json');
```
Comment (karlhorky) [inline bin/install.js:38]:
```suggestion
  console.log('✅ Configured package manager in package.json for Netlify');
```

File: bin/install.js (hunk 3 above)
Comment (ProchaLu) [reply bin/install.js:38]: Added a line and a comment in this commit
[6e09664](https://github.com/upleveled/eslint-config-upleveled/pull/228/commits/6e09664d93ed24b94952e9a214c0315d03dcea37)
Also edited the version number in the comment
[c8f207f](https://github.com/upleveled/eslint-config-upleveled/pull/228/commits/c8f207fb7e43198b102a9fd5ab719cc4694d30a6)

**Comment by karlhorky** [comment]
I reverted this now because [Netlify now uses 8.9.2](https://answers.netlify.com/t/upgrade-netlify-build-to-pnpm-v8/90570/5):

- https://github.com/upleveled/eslint-config-upleveled/commit/0019e173c9022d78c1598f7551d8127436e4715f

**Pull request merged**